scripts/commands/updateApiDocs.ts

```diff
@@ -336,6 +336,14 @@ async function convertHtmlToMarkdown(
     await writeFile(path, result.markdown);
   }
 
+  if (pkg.name == "qiskit") {
```

Comment:
It's an issue for Provider and Runtime too:

https://github.com/Qiskit/documentation/blob/0b60acf1a5da5c0653fb3ed4bc1dc04700e5e1f7/docs/api/qiskit-ibm-provider/_toc.json#L9

I suspect that instead the fix is going to be inside `generateToc`, that it should special-case `/index`. 

I also remember wondering if the fix belongs in https://github.com/Qiskit/documentation/blob/main/scripts/lib/sphinx/renameUrls.ts, but I think I convinced myself a few weeks ago that no. It's important we keep the `index` file name there so that we create a file called `index.md`. We only need to not use `/index` in `generateToc`.

Reply:
I removed the changes in `updateApiDocs.ts` and updated the `generateToc.ts` file to not use /index in any URL.